runtime/convertValues.go

```diff
@@ -423,6 +448,22 @@ func importCompositeValue(
 		)
 	}
 
+	var builtinType *sema.BuiltinCompositeType
+	switch qualifiedIdentifier {
+	case sema.PublicKeyType.QualifiedString():
+		builtinType = sema.PublicKeyType
+	case sema.AccountKeyType.QualifiedString():
+		builtinType = sema.AccountKeyType
+	case sema.HashAlgorithmType.QualifiedString():
+		builtinType = sema.HashAlgorithmType
+	case sema.SignatureAlgorithmType.QualifiedString():
+		builtinType = sema.SignatureAlgorithmType
+	}
```

Comment:
Maybe just declare them in a map and perform the lookup there?

Also, I think this switch should only be used if the location is native? (or once removed, I guess is nil). Currently this code would result in type confusion: if a value is imported that happens to have the same name as a built-in type, but is actually in e.g. an address location (from a deployed contract), then the built-in type is incorrectly returned.

Reply:
I think both issues will go away if we do the unification I mentioned here: https://github.com/onflow/cadence/pull/633#discussion_r586100296